.github/workflows/test-osbuild-composer-integration.yml

```diff
@@ -0,0 +1,161 @@
+        run: git config --global --add safe.directory "$(pwd)"
+
+      - name: Update the osbuild/images reference to the merge base with main
+        env:
+          merge_base_sha: ${{ github.event.pull_request.base.sha }}
+        run: |
+          cd osbuild-composer
+          go mod edit -replace github.com/osbuild/images=github.com/osbuild/images@$merge_base_sha
+          ./tools/prepare-source.sh
+
+      - name: Run unit tests (merge base)
+        working-directory: osbuild-composer
+        id: tests-base
+        # This step will not fail if the test fails, but it will write the
+        # failure to GITHUB_OUTPUT
+        run: |
+          if go test -v -race ./...; then
+            echo "base_test=1" >> $GITHUB_OUTPUT
+          else
+            echo "base_test=0" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Update the osbuild/images reference to the PR HEAD
+        env:
+          pr_head: ${{ github.event.pull_request.hase.sha }}
+        # if the base tests failed, there's no need to run the PR HEAD tests
+        if: steps.tests-base.outputs.base_test == 1
+        # Restore and clean the checkout and replace the dependency again using
+        # images from the checkout above
+        run: |
+          cd osbuild-composer
+          git restore .
+          git clean -xfd .
+          go mod edit -replace github.com/osbuild/images=../images
+          ./tools/prepare-source.sh
+
+      - name: Run unit tests (PR HEAD)
+        id: tests-pr
+        working-directory: osbuild-composer
+        # if the base tests failed, there's no need to run the PR HEAD tests
+        if: steps.tests-base.outputs.base_test == 1
+        # This step will not fail if the test fails, but it will write the
+        # failure to GITHUB_OUTPUT
+        run: |
+          if go test -v -race ./...; then
+            echo "pr_test=1" >> $GITHUB_OUTPUT
+          else
+            echo "pr_test=0" >> $GITHUB_OUTPUT
+          fi
+
+  post-results:
+    name: "Post notice"
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    needs: unit-tests
+    steps:
+      - name: Add comment (breakage)
+        uses: mshick/add-pr-comment@v2
+        repo-token: ${{ secrets.SCHUTZBOT_GITHUB_ACCESS_TOKEN }}
+        if: needs.unit-tests.outputs.base_test == 1  &&  needs.unit-tests.outputs.pr_test == 0
+        with:
+          issue: ${{ github.event.pull_request.number }}
+          message: |
+            This PR changes the images API or behaviour causing integration failures with osbuild-composer. The next update of the images dependency in osbuild-composer will need work to adapt to these changes.
+
+            This is simply a notice. It will not block this PR from being merged.
+
+      - name: Update comment (fixed)
+        uses: mshick/add-pr-comment@v2
+        repo-token: ${{ secrets.SCHUTZBOT_GITHUB_ACCESS_TOKEN }}
+        if: needs.unit-tests.outputs.pr_test == 1
```

Comment:
I'm not too familiar with github actions but will this "crash" somehow if `needs.unit-tests.outputs.base_test == 0` as `needs.unit-tests.outputs.pr_test` might not be defined at all due to ` if: steps.tests-base.outputs.base_test == 1`

Reply:
No, it will default to an empty string.  At least in my testing in the experiments repo it did.